### PR TITLE
Support Matrix

### LINKED ISSUE #1052: Support Matrix
Cookiecutter Django is loaded with features. Not all of them work on every environment out of the box, or we don't have docs for it.

It'd great if we had a matrix that shows what feature is supported on what environment:

# Development

| | Vanilla Python (mac)  | Vanilla Python (Linux)   | Vanilla Python (windows)   | Docker (mac)  | Docker (windows)   | Docker (Linux)
|---|---|---|---|---|---|---|
| postgres |   |   |   |   | ||
| mailhog |   |   |   |   | ||
| grunt |   |   |   |   | ||
| gulp |   |   |   |   | ||
| pycharm |   |   |   |   | ||

# Production

| | Ubuntu  | Docker  | Heroku  | Elastic Beanstalk  | Python Anywhere   |
|---|---|---|---|---|---|
| postgres |   |   |   |   | ||
| redis |   |   |   |   | ||
| celery |   |   |   |   | ||
| grunt |   |   |   |   | ||
| gulp |   |   |   |   | ||
| whitenoise |   |   |   |   | ||
| static files |   |   |   |   | ||
| lets encrypt |   |   |   |   | ||
| compressor |   |   |   |   | ||

Is someone able to help out and fill (parts) of this matrix?

### PR DESCRIPTION
**Pull request by webyneter**
Closes #1052.

### CODE REVIEW

**Comment by jayfk** [comment]
Is there a way to see the updated docs in a human readable form? :D

**Comment by webyneter** [comment]
@jayfk you mean, in Read the Docs, built for this particular branch? I don't have access to make it work, unfortunatelly.

**Comment by webyneter** [comment]
@jayfk for now, it's just a plain-text table in rst doc.

**Comment by jayfk** [comment]
> @jayfk for now, it's just a plain-text table in rst doc.

I was looking at this file and that confused me: https://github.com/pydanny/cookiecutter-django/pull/1290/files#diff-af0dc7a1d9e7c3ddbf4eed7a1d29794b

**Comment by webyneter** [comment]
@jayfk:
"+Environments
+------------
+
+.. Generated via http://www.tablesgenerator.com/text_tables
+.. Header rows and columns' thick borders set manually."

**Comment by webyneter** [comment]
Just a serialized representation of the table allowing for further edits.

**Comment by jayfk** [comment]
> @jayfk:
> "+Environments
> +------------
> +
> +.. Generated via http://www.tablesgenerator.com/text_tables
> +.. Header rows and columns' thick borders set manually."

Generators ¯\\__(ツ)_/¯

**Comment by webyneter** [comment]
¯\__(ツ)_/¯

File: docs/support-matrix.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+.. _`WhiteNoise`: https://github.com/evansd/whitenoise
+.. _`Celery`: https://github.com/celery/celery
+.. _`MailHog`: https://github.com/mailhog/MailHog
+.. _`Sentry`: https://github.com/getsentry/sentry
+.. _`Opbeat`: https://github.com/opbeat/opbeat_python
+.. _`Compressor`: https://github.com/django-compressor/django-compressor
+.. _`Gulp`: https://github.com/gulpjs/gulp
+.. _`Grunt`: https://github.com/gruntjs/grunt
+.. _`Heroku`: https://www.heroku.com/
+.. _`Elastic Beanstalk`: https://aws.amazon.com/elasticbeanstalk/
```
Comment (browniebroke) [inline docs/support-matrix.rst]: This should be removed now

**Comment by browniebroke** [comment]
Could you update your branch from `master`? Would be good to get this in!

File: docs/support-matrix.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+* **Bare bones**: native Python setup implying the use of `virtualenv`_ and friends (`pyenv`_, `pyenv-virtualenv`_, `virtualenvwrapper`_, `pyenv-virtualenvwrapper`_, `pipenv`_ etc.)
+* **Docker**: Docker-powered setup implying the use of `Docker`_, `Docker Compose`_ etc.
+
+.. _`virtualenv`: https://github.com/pypa/virtualenv
+.. _`pyenv`: https://github.com/pyenv/pyenv
+.. _`pyenv-virtualenv`: https://github.com/pyenv/pyenv-virtualenv
+.. _`virtualenvwrapper`: https://virtualenvwrapper.readthedocs.io/en/stable/
+.. _`pyenv-virtualenvwrapper`: https://github.com/pyenv/pyenv-virtualenvwrapper
+.. _`pipenv`: https://github.com/kennethreitz/pipenv
```
Comment (browniebroke) [inline docs/support-matrix.rst]: This is now https://github.com/pypa/pipenv

**Pull request closed without merging**